Content.Server/Imperial/Radio/JobPlayerAndColor.cs

```diff
@@ -0,0 +1,173 @@
+        return job;
+    }
+
+	/// <summary>
+    /// Для EnityUid ищется карта в руках или в пда. Если карта нашлась, мы смотрим какая там работа и достаём её
+    /// </summary>
+    /// <param name="uid"> uid игрока, у которого мы ищем карту</param>
+    /// <returns></returns>
+    public string? GetJobPlayer(EntityUid uid)
+    {
+        // Проверяем нашлась ли карта и какое id у карты
+        if (_idCardSystem.TryFindIdCard(uid, out var id))
+        {
+            // Мы сохраняем работу
+            string? playerJob = id.Comp.LocalizedJobTitle;
+            // Если работа нашлась верно, мы начинаем основной процесс
+            if (playerJob != null)
+            {
+                // Делаем начало должности с заглавной буквы и сохраняем в playerJob
+                playerJob = char.ToUpper(playerJob[0]) + playerJob.Substring(1);
+                // Заменяем тире, дефисы и минусы на пробелы. Эти символы при форматировании ломают вывод сообщения в рацию
+                playerJob = Regex.Replace(playerJob, @"[-–—−]", " ");
+                // Уберём лишние символы "!?", которые могут ломать в целом вывод сообщения в радио канале
+                playerJob = Regex.Replace(playerJob, @"[^a-zA-Zа-яА-ЯёЁ ]", "");
+                // Переводим должность через словарь. Если перевода нет, playerJob не меняется
+                playerJob = TranslateJob(playerJob);
+                return playerJob.Trim();
+
+            }
+
+
+        }
+        // Если работы нет, то возвращается должность "Неизвестно"
+        return "Неизвестно";
+    }
+
+    /// <summary>
+    /// Метод, который отвечает за подбор цвета для должности. Используется словарь, который работает по О(1), что быстрее if и подобного
+    /// </summary>
+    /// <param name="jobPlayer"> Нужно получить из метода GetJobPlayer. Или же вставляйте сюда свою string? работу, но нужно привести её к нижниму регистру для словаря. Метод ваш_стринг.ToLower()</param>
+    /// <returns></returns>
+    public string? GetColorPlayer(string? jobPlayer)
+    {
+        // Проверка. Работаем только тогда, когда работа была определена успешно
+        if (jobPlayer != null)
+        {
+            // Преобразуем jobPlayer к нижнему регистру для поиска в словаре
+            string normalizedJob = jobPlayer.ToLower();
+
+            // Выполняется по О(1). Если вдруг будет очистка памяти и словарь удалится, то мы будем уверены что он снова инициализируется
+            IsDictionaryEmpty();
+
+            // Проверка наличия должности в словаре
+            if (SaveColorForJob.ContainsKey(normalizedJob))
+            {
+                // Если должность найдена в словаре, возвращаем соответствующий цвет
+                string color = SaveColorForJob[normalizedJob];
+                return color;
+            }
+            else
+            {
+                // Если должность не найдена в словаре, возвращаем дефолтный цвет
+                return "lime";
+            }
```

Comment:
Инвёрс условия:
```suggestion
            // Если должность не найдена
            if (!SaveColorForJob.ContainsKey(normalizedJob)) return "lime";
            
            // Возвращаем цвет должности
            string color = SaveColorForJob[normalizedJob];
            return color;
```

(А вообще лучше сразу избавиться от двойного поиска:
```suggestion
return SaveColorForJob.TryGetValue(normalizedJob, out var color) ? color : "lime"
```
)

Reply:
Отсутствие обработки ошибок будет приводить к тому, что ошибки не будут обрабатываться. Так же в jobPlayer мы уже задаём значения для тех работ, которые невозможно найти или их нет

Закрываю

Reply:
Мой изменённый код полностью идентичен твоему коду и работает точно так же, НО не совершает поиск по словарю дважды (что как раз таки и вредит оптимизации)